### PR TITLE
Add small test for v15 backwards compatibility

### PR DESCRIPTION
**Pull request by kzh**

### CODE REVIEW

File: read_test.go [hunk 1]
```diff
@@ -98,3 +103,186 @@ func TestReadWriteNames(t *testing.T) {
...
+
+	qs := []query.Q{
+		query.NewAnd(&query.Substring{Pattern: "func main", Content: true}, &query.Substring{Pattern: "zoekt/main.go", FileName: true}),
+		query.NewAnd(&query.Regexp{Regexp: mustParseRE("^package"), Content: true}, &query.Substring{Pattern: "zoekt/main.go", FileName: true}),
+		query.NewAnd(&query.Symbol{&query.Substring{Pattern: "shard"}}, &query.Substring{Pattern: "zoekt/main.go", FileName: true}),
+		query.NewAnd(&query.Symbol{&query.Regexp{Regexp: mustParseRE("shard$")}}, &query.Substring{Pattern: "zoekt/main.go", FileName: true}),
+	}
+
+	for _, tc := range tcs {
+		shard, err := loadShard("test-index/" + tc.file)
```
Comment (keegancsmith) [inline read_test.go]: In go we call test data directories like this `testdata`.

**Review by keegancsmith — CHANGES_REQUESTED** [review]
These files are huge. Can we instead just index a very simple repo containing one file?

```
$ du -h test-index/
4.7M    test-index/
```

**Comment by keegancsmith** [comment]
@kzh ping?

**Comment by kzh** [comment]
Hey @keegancsmith, I updated the test to make use of a golden file for validation. Sorry, I got a little sidetracked with the performance testing on symbol search.

File: read_test.go [hunk 2]
```diff
@@ -98,3 +104,94 @@ func TestReadWriteNames(t *testing.T) {
...
+	}
+
+	golden := "testdata/golden/TestReadSearch.golden"
+	f, err := os.Open(golden)
+	if err != nil {
+		t.Fatalf("error opening golden file %s", golden)
+	}
+	defer f.Close()
+
+	if err := json.NewDecoder(f).Decode(&want); err != nil {
```
Comment (keegancsmith) [inline read_test.go]: don't we want a different golden file per shard? ie I'd suspect we should test shards without symbols.

File: read_test.go [hunk 3]
```diff
@@ -98,3 +104,94 @@ func TestReadWriteNames(t *testing.T) {
...
+
+func TestReadSearch(t *testing.T) {
+	var want []struct {
+		FormatVersion  int
+		FeatureVersion int
+		FileMatches    [][]FileMatch
+	}
+
+	golden := "testdata/golden/TestReadSearch.golden"
+	f, err := os.Open(golden)
```
Comment (keegancsmith) [inline read_test.go]: You can avoid the file you need to close via something like

```go
if buf, err := ioutil.ReadFile(golden); err != nil {
  t.Fatal(err)
} else if err = json.Unmarshal(buf, &want); err != nil {
  t.Fatal(err)
}
```

**Review by keegancsmith — APPROVED** [review]
Two changes before landing:

- Can you include a way to update the golden file. Normally one adds a `-update` flag. EG see how we do it in the sourcegraph repo or the golang repo https://sourcegraph.com/search?q=r:%5Egithub.com/%28sourcegraph/sourcegraph%7Cgolang/go%29%24+flag%5C.%5BA-Z%5D+%28golden%7Cupdate%29+f:test.go
- Marshal the golden files with indentation. Will make it easier to review changes to them.

**Review by keegancsmith — APPROVED** [review]

**Pull request merged**